### PR TITLE
[15.0][MIG] website_sale_product_assortment: Migration to version 15.0

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
cc@Tecnativa TT39034

@chienandalu @CarlosRoca13 please review!

### CODE REVIEW

File: website_sale_product_assortment/controllers/website_sale.py [hunk 1]
```diff
@@ -51,18 +50,43 @@ def product(self, product, category="", search="", **kwargs):
...
+    #     """
+    #     res = super()._get_search_domain(
+    #         search, category, attrib_values, search_in_description=search_in_description
+    #     )
+    #     allowed_product_ids, assortment_restriction = self._get_products_allowed()
+    #     if assortment_restriction:
+    #         return expression.AND(
+    #             [res, [("product_variant_ids", "in", list(allowed_product_ids))]]
+    #         )
+    #     return res
```
Comment (CarlosRoca13) [inline website_sale_product_assortment/controllers/website_sale.py]: Ops! remove this

File: website_sale_product_assortment/controllers/website_sale.py [hunk 2]
```diff
@@ -51,18 +50,43 @@ def product(self, product, category="", search="", **kwargs):
...
+        attrib_values=None,
+        pricelist=None,
+        min_price=0.0,
+        max_price=0.0,
+        conversion_rate=1,
+        **post
     ):
-        """Overriding _get_search_domain method to avoid show product templates that
-        has all their variants not allowed to be shown.
-        """
```
Comment (CarlosRoca13) [inline website_sale_product_assortment/controllers/website_sale.py]: Addapt the comment and keep it

File: website_sale_product_assortment/controllers/website_sale.py (hunk 1 above)
Comment (stefan-tecnativa) [reply website_sale_product_assortment/controllers/website_sale.py]: A small oversight

File: website_sale_product_assortment/controllers/website_sale.py (hunk 2 above)
Comment (stefan-tecnativa) [reply website_sale_product_assortment/controllers/website_sale.py]: Done

**Review by CarlosRoca13 — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
Test and code review. Good job :+1: !

**Pull request merged**